### ISSUE TITLE
Since v25, Legacy SMS Support No Longer Working With Samsung Galaxy S5, S6

### ISSUE THREAD

**Issue by JohannPopper** [andyholmes/gnome-shell-extension-gsconnect#650]
Testing the latest GSConnect with my old phones.

Version 24 works as expected. With KDE Connect 1.13.2 & GSConnect v25+, however, Legacy SMS Support doesn't seem to work at all anymore. Whether it's switched active or not, I'm not seeing the doubled legacy messages in my notifications anymore, and just as if Legacy is switched off, clicking on the single standard notification just brings up my address book, not the texting window, as was the case before the Legacy SMS Support feature was added last spring (thanks). Could there be a bug causing this? 

**Support Log**
The following is all I see in the log when I receive an sms, identical whether Legacy SMS is active or not since v25 on both S5 and S6. 

```
Oct 02 12:13:37 org.gnome.Shell.Extensions.GSConnect[24609]: [/service/protocol/core.js:send:321]: Galaxy S5: {
                                                               "id": 157********51,
                                                               "type": "kdeconnect.notification.request",
                                                               "body": {
                                                                 "cancel": "sms|+187******85"
                                                               }
                                                             }
Oct 02 12:13:37 org.gnome.Shell.desktop[900]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x30073b6
```

 - **GSConnect version:** v26
   - **Installed from:** GNOME Extensions Website
 - **GNOME/Shell version:** 3.32.1
 - **Distro/Release:** Solus 4

 - **Paired Device(s):** Samsung Galaxy S5
 - **KDE Connect app version:** 1.13.2

**Additional Notes:**
The KDE Connect App recently updated on my phone also.

**Comment by philipraets** [comment]
I can confirm this with a samsung galaxy note 4 (N910F) Android 6.0.1, KDEconnect 1.13.2 and GSconnect (latest from master)

**Comment by andyholmes** [comment]
Thanks for reporting, I think these problems came up when the SMS code was refactored to support the new protocol changes.

I think most of the critical problems should be fixed in `master` now, but there are a few more quirks to fix, so I'll leave this open until I get those sorted out.

**Comment by andyholmes** [comment]
Okay, I think I've solved all the problems now as of 889c43c.

I'm going to leave this open until someone who regularly uses the feature can confirm all is well, while I port them over to the notification dialog which is quite similar :)

**Comment by JohannPopper** [comment]
Hi, Andy. Jeebus, you're efficiently charitable. I just tested master. Works perfectly again (although I first had to delete the old v26 cache/settings manually after install, otherwise GSConnect wouldn't recognize the phone even after a reboot -- no more Delete button in Advanced settings). 

Thank you! If you don't have a <buymeacoffee> account or something, you definitely should & link it to your github. GSConnect has these amazing essential features upstream KDE doesn't yet.
